src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -23,38 +37,189 @@
+  private NetworkTableEntry IDelayTable = shooterTab.add("Current I delay",0).withPosition(2, 4).getEntry(); //Delay before the CDS deliver the ball for the PID to stablize the speed
+  private double MaxOutputConstant;
+  private double MinOutputConstant;
+  private ShooterConfig[] DistanceArray;
+
+
+
+
+
+  
+  
+  //private NetworkTableEntry PID_F = shooterTab.addPersistent("PID F", Constants.Shooter.).withPosition(3, 2).getEntry();
+  //private NetworkTableEntry PID_Izone = shooterTab.addPersistent("PID I Range", Constants.Shooter.).withPosition(4, 2).getEntry();
+  //private NetworkTableEntry PID_MaxOutput = shooterTab.addPersistent("PID Peak Output", Constants.Shooter.d).withPosition(5, 2).getEntry();
+
+
+
+
+
+
+
+
   public ShooterSubsystem() {
-    
+    //SmartDashboard.putNumber("RPMIN", RPMIN);
     aimMode = 4;
-    cargo_motorController = new MotorController("Shooter Cargo", Constants.shooterCargoID);
-    shooter_motorController = new MotorController("Shooter", Constants.shooterID, 40, true);
+    cargo_motorController = new MotorController("Shooter Cargo", Constants.Shooter.shooterCargoID,40,true);
+    kCargoController = cargo_motorController.getPID();
+    kCargoEncoder = cargo_motorController.getEncoder();
+    shooter_motorController = new MotorController("Shooter", Constants.Shooter.shooterID, 40, true);
     KShooterController = shooter_motorController.getPID();
     KShooterEncoder = shooter_motorController.getEncoder();
-    KShooterController.setP(5e-4);
-    KShooterController.setI(6e-7);
-    KShooterController.setD(0.0);
+    
+    DistanceArray[0] = new ShooterConfig(0,0, 0);
+    DistanceArray[1] = new ShooterConfig(0, 0, 0);
+    DistanceArray[2] = new ShooterConfig(0,0, 0);
+    DistanceArray[3] = new ShooterConfig(0,0, 0);
+    DistanceArray[4] = new ShooterConfig(0,0, 0);
+    DistanceArray[5] = new ShooterConfig(0,0, 0);
+    DistanceArray[6] = new ShooterConfig(0,0, 0);
+    DistanceArray[7] = new ShooterConfig(0,0, 0);
+    DistanceArray[8] = new ShooterConfig(0,0, 0);
+    DistanceArray[9] = new ShooterConfig(0,0, 0);
+    DistanceArray[10] = new ShooterConfig(0,0, 0);
+    DistanceArray[11] = new ShooterConfig(0,0, 0);
+    DistanceArray[12] = new ShooterConfig(0,0, 0);
+    DistanceArray[13] = new ShooterConfig(0,0, 0);
+    DistanceArray[14] = new ShooterConfig(0,0, 0);
+    DistanceArray[15] = new ShooterConfig(0,0, 0);
+    DistanceArray[16] = new ShooterConfig(0,0, 0);
+    DistanceArray[17] = new ShooterConfig(0,0, 0);
+    DistanceArray[18] = new ShooterConfig(0,0, 0);
+    DistanceArray[19] = new ShooterConfig(0,0, 0);
+    //TODO:FIll lookup table
+    
 
-    KShooterController.setOutputRange(0, 1);
-    /*
-    hood_motorController = new MotorController("Hood", Constants.hoodID);
-    KHoodController = hood_motorController.getPID();
-    KHoodEncoder = shooter_motorController.getEncoder();*/
+
+
+
+    //KShooterController.setP(5e-4);
```

Comment:
What's going on here?  Does the setP need to be in code, or not.  Let's figure that out, and remove or note why it's commented out.

Reply:
( Same for the other PID controls below. )

Reply:
talked to Jim:  Hood still needs to be set up, deleted pid commands